### DISCUSSION TITLE
Jetson Orin Nano 4GB (P3767-0004) Unable to access the system

### DISCUSSION THREAD

**Discussion by Links17** [OE4T/tegra-demo-distro#271] [Q&A] [answered]
git branch:
kirkstone
cmd:
source ../setup-env --machine p3768-0000-p3767-0004
bitbake demo-image-base
Compile Successful Image,But after I initrd-flash, I can't get into the system.
![image](https://github.com/OE4T/tegra-demo-distro/assets/61582001/83059dce-0c47-47e3-a1b2-cb6ef88b3012)

**Comment by madisongh** [comment]
I saw this recently, and it was because the flashing wasn't completely successful.  If you are not in the `disk` group, the `initrd-flash` script isn't able to partition the external drive.  It reports an error when that happens, but the reported final status for flashing still shows up as successful.  I need to fix that.

Make sure you're in the `disk` group (you can check that with the `groups` command), or re-run the `initrd-flash` script under `sudo`.  For this particular device, since flashing the boot firmware looks like it worked OK, you can use `sudo ./initrd-flash --skip-bootloader` to just re-flash the NVMe drive.

**Comment by Links17** [comment]
Thank you very much for your answer, after initrd-flash, is it still necessary to execute doflash? Forgive my ignorance...

**Reply by madisongh** [reply]
No.  It's one or the other, not both.

**Comment by Links17** [comment]
At that time, after running sudo . /initrd-flash --skip-bootloader, but still can not enter the system.
<img width="640" alt="image" src="https://github.com/OE4T/tegra-demo-distro/assets/61582001/a66f6c1c-3c48-4e1c-a957-e0cc34658d3a">
![image](https://github.com/OE4T/tegra-demo-distro/assets/61582001/14022599-4429-4797-8158-17d1b2d7f8a9)

**Reply by madisongh** [reply]
The `No partition definitions found in initrd-flash.xml` message is the key here.  It looks like we need to fix the machine configuration file for the Nano 4GB.

Add the following lines to your `$BUILDDIR/conf/local.conf` file:
```
TNSPEC_BOOTDEV_DEFAULT:p3768-0000-p3767-0004 = "mmcblk0p1"
TNSPEC_BOOTDEV:p3768-0000-p3767-00004 = "nvme0n1p1"
```
then re-bitbake your image and try the `initrd-flash` again (under sudo if needed).

**Comment by Links17** [comment]
Thanks for your patience, it's already under construction!

**Reply by madisongh** [reply]
Let me know if this works.  If so, I'll fix the machine config file so you won't need to add those extra liness.

**Reply by madisongh** [reply, marked as answer]
Sorry, I think I had a typo in that second line to add -- an extra `0` in the `0004` part of the variable suffix.  initrd-flash thinks you're trying to flash to `mmcblk0p1` instead of `nvme0n1p1`.
That second line should be

```
TNSPEC_BOOTDEV:p3768-0000-p3767-0004 = "nvme0n1p1"
```

Sorry about that

**Comment by Links17** [comment]
The situation looks better than before, but there's a new error.
<img width="1047" alt="image" src="https://github.com/OE4T/tegra-demo-distro/assets/61582001/99e2fdb3-3244-46dc-afc4-c96e2922788a">

**Comment by Links17** [comment]
Thank you very much! It's already running, thanks a lot for the answer!

**Reply by madisongh** [reply]
I've updated the `p3768-0000-p3767-0004.conf` file to fix the issue, so next time you update the layer, you shouldn't need the extra lines in your local.conf any more.
